src/sql/workbench/contrib/notebook/browser/turndownPluginGfm.ts

```diff
@@ -99,12 +99,12 @@ rules['table'] = {
 		// Ensure there are no blank lines
 		content = content.replace('\n\n', '\n');
 		// if the headings are empty, add border line and headings to keep table format
-		if (node.tHead.innerText === '') {
+		if (node.tHead?.innerText === '') {
```

Comment:
You want to do this logic if it doesn't have a thead too though right? So it should be `if (node.tHead?.innerText ?? '' === '') {`

Reply:
Ok so after discussion with Chris I think this is fine as a fix for the immediate issue - but I think there's something not quite right here. We should only be in this function (based on the filter) if we detect that the first row is in a thead element OR is all th's (outside of a thead). But there should always be a row of th elements as the first row. 

And based on what I see something like this should be valid HTML (albeit a bit odd) but wouldn't be handled correctly here because it'd add a new row of blank values on top of the existing header row : 

```
<table>
   <thead></thead>
   <tr><th>col1</th></tr>
   <tr><td>val1</td></tr>
</table>
```

@barbaravaldez FYI

Reply:
We chatted about this offline and did some analysis here. Going to keep existing implementation to fix this problem.